ros/lib/kessel/kessel_shared.py

```diff
@@ -0,0 +1,99 @@
+    This is a shared cache used by both the gRPC singleton and KesselClient instances.
+    Returns:
+        OAuth2ClientCredentials object or None if creation fails
+    """
+    global _AUTH_CREDENTIALS
+    if _AUTH_CREDENTIALS is None:
+        from ros.lib.config import create_kessel_oauth2_credentials
+        LOG.debug("Creating shared OAuth2 credentials cache (first time)")
+        _AUTH_CREDENTIALS = create_kessel_oauth2_credentials()
+    return _AUTH_CREDENTIALS
+
+
+def get_kessel_stub(host):
+    """
+    Get or create a Kessel gRPC stub with OAuth2 authentication.
+
+    Args:
+        host: Kessel service host and port
+
+    Returns:
+        KesselInventoryServiceStub instance
+    """
+    global _GRPC_CHANNEL, _GRPC_STUB
+
+    if _GRPC_STUB is None:
+        try:
+            from ros.lib.config import (
+                KESSEL_AUTH_CLIENT_ID,
+                KESSEL_AUTH_CLIENT_SECRET,
+                KESSEL_AUTH_OIDC_ISSUER,
+                KESSEL_INSECURE
+            )
+
+            if KESSEL_AUTH_CLIENT_ID and KESSEL_AUTH_CLIENT_SECRET and KESSEL_AUTH_OIDC_ISSUER:
+                try:
+                    from kessel.inventory.v1beta2 import ClientBuilder
+
+                    LOG.debug("Creating OAuth2 authenticated Kessel connection")
+
+                    # Create OAuth2 client credentials using shared cache
+                    auth_credentials = get_cached_kessel_auth_credentials()
+                    if not auth_credentials:
+                        raise Exception("Failed to create OAuth2 credentials")
+
+                    # Create authenticated stub and channel using ClientBuilder
+                    LOG.debug(f"Creating authenticated Kessel client for {host}")
+
+                    if KESSEL_INSECURE:
+                        # Use local channel credentials for insecure connections
+                        LOG.info("Using insecure channel credentials with OAuth2 authentication")
+                        _GRPC_STUB, _GRPC_CHANNEL = (
+                            ClientBuilder(host)
+                            .oauth2_client_authenticated(auth_credentials, grpc.local_channel_credentials())
+                            .build()
+                        )
+                    else:
+                        # Use default secure credentials for TLS connections
+                        LOG.info("Using secure TLS channel credentials with OAuth2 authentication")
+                        _GRPC_STUB, _GRPC_CHANNEL = (
+                            ClientBuilder(host)
+                            .oauth2_client_authenticated(auth_credentials)
+                            .build()
+                        )
+                    LOG.info("Successfully created authenticated Kessel client")
+
+                except ImportError:
+                    LOG.warning("kessel_sdk OAuth2 authentication not available, falling back to insecure connection")
```

Comment:
This shouldn't happen, if it does, there are some dependencies missing

Reply:
Could you update the [kessel-sdk](https://github.com/RedHatInsights/ros-backend/commit/f9bf4a7d25fb2683322910b88e28192ccb423817) to use the latest sdk with the auth optional?

It should be more or less: 

```
"kessel-sdk[auth]>=1.5.0",
```